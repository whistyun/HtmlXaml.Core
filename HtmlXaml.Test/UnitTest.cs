
using ApprovalTests;
using ApprovalTests.Reporters;
using HtmlXaml.Core;
using NUnit.Framework;
using System.Linq;
using System.Threading;
using System.Windows.Documents;

namespace HtmlXaml.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest
    {
        [Test]
        [Apartment(ApartmentState.STA)]
        public void Button()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void CodeBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void InlineCode()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Input()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void List()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Progres()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalInline()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
