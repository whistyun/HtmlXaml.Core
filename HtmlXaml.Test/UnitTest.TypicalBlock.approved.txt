<?xml version="1.0" encoding="utf-16"?>
<FlowDocument xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
  <Paragraph Tag="Heading1">
    <Bold Tag="BoldSpan">Heading1</Bold> Title</Paragraph>
  <Paragraph Tag="Heading2">
    <Italic Tag="ItalicSpan">Sub</Italic> Titile</Paragraph>
  <Paragraph Tag="Heading3">H3_Title</Paragraph>
  <Paragraph Tag="Heading4">H4_Title</Paragraph>
  <Paragraph Tag="Heading5">H5_Title</Paragraph>
  <Paragraph Tag="Heading6">H6_Title</Paragraph>
  <Section Tag="Address">
    <Paragraph>Written by <Hyperlink Command="{x:Null}" CommandParameter="mailto:webmaster@example.com" Tag="Hyperlink">Jon Doe</Hyperlink>.<LineBreak />simewhere, someland, somecountry</Paragraph>
  </Section>
  <Section Tag="Article">
    <Paragraph Tag="Heading3" xml:space="preserve">Article         Element</Paragraph>
    <Paragraph xml:space="preserve">description of         this element...</Paragraph>
  </Section>
  <Section Tag="Aside">
    <Paragraph Tag="Heading4" xml:space="preserve">ASide         Element</Paragraph>
    <Paragraph>このタグに関する説明を書いてください・・・</Paragraph>
  </Section>
  <Section Tag="Blockquote">
    <Section Tag="Aside">
      <Paragraph Tag="Heading4" xml:space="preserve">ASide             Element</Paragraph>
      <Paragraph>このタグに関する説明を書いてください・・・</Paragraph>
    </Section>
    <Paragraph>hm?</Paragraph>
  </Section>
  <Section TextAlignment="Center" Tag="Center">
    <Paragraph Tag="Heading5">Center Element</Paragraph>
    <Paragraph>center tag center tag center tag...</Paragraph>
  </Section>
  <Paragraph Tag="Heading6">H6_Title</Paragraph>
  <Paragraph>abcdefghi</Paragraph>
  <Paragraph>jklmnopqr</Paragraph>
  <Paragraph>stuvwxyz</Paragraph>
  <Paragraph>abcdefghi</Paragraph>
  <Paragraph>jklmnopqr<Run>stuvwxyz</Run></Paragraph>
</FlowDocument>
